torch/testing/_internal/common_methods_invocations.py

```diff
@@ -578,22 +578,85 @@ def sample_inputs_linalg_vector_norm(op_info, device, dtype, requires_grad):
 
     return inputs
 
+def sample_inputs_mm(op_info, device, dtype, requires_grad):
+    args_list = (
+        ((S, M), (M, S)),
+    )
+    inputs = list(SampleInput(make_tensor(first_shape, device, dtype,
+                                          requires_grad=requires_grad),
+                              args=(make_tensor(second_shape, device, dtype,
+                                    requires_grad=requires_grad),))
+                  for first_shape, second_shape in args_list)
+    return inputs
+
+def sample_inputs_add(op_info, device, dtype, requires_grad):
+    inputs = []
+    args_list = (
+        ((S, S, S), (S, S, S)),
+        ((S, S, S), (S, S)),
+        ((S, S), (S, S, S)),
+        ((S, 1, S), (M, S)),
+        ((), ()),
+        ((S, S, S), ()),
+        ((), (S, S, S)),
+    )
+    inputs = list(SampleInput(make_tensor(first_shape, device=device, dtype=dtype,
+                                          requires_grad=requires_grad),
+                              args=(make_tensor(second_shape, device=device, dtype=dtype,
+                                    requires_grad=requires_grad),))
+                  for first_shape, second_shape in args_list)
+    if dtype in [torch.bfloat16, torch.float16, torch.float32, torch.float64]:
+        second_args_list = (
+            ((S, S, S), [3.14]),
+            ((), [3.14]),
+        )
+        more_inputs = list(SampleInput(make_tensor(first_shape, device=device, dtype=dtype,
+                                                   requires_grad=requires_grad),
+                                       args=(torch.tensor(additive, device=device, dtype=dtype,
+                                                          requires_grad=requires_grad),))
+                           for first_shape, additive in second_args_list)
+        inputs = [*inputs, *more_inputs]
+    elif dtype in [torch.complex64, torch.complex128]:
+        inputs.append(SampleInput(make_tensor((S, S, S), device, dtype,
+                                              requires_grad=requires_grad),
+                                  args=(torch.tensor([3.14j], device=device, dtype=dtype,
+                                                     requires_grad=requires_grad),)))
+    return inputs
+
+def sample_inputs_addmm_non_fusible_nodes(op_info, device, dtype, requires_grad):
+    args_tuple = (
+        ((S, M), (S, S), (S, M)),
+        ((1,), (S, S), (S, M)),
```

Comment:
Same question about how this works with the inplace varaint as for mm

Reply:
@kshitij12345's answer is correct. Let's get that test fixed and then we can sort out these OpInfos.

Reply:
@mruberry: Oops, I probably should've noticed something was amiss.

Reply:
@imaginary-person I don't think so. This is a pretty well hidden interaction.